### PR TITLE
[AIR][CUJ] Add GPU bench prediction benchmark 

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

as titled.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/air_tests/air_benchmarks/workloads/gpu_batch_prediction.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    print(test_output_json)
+
+
+if __name__ == "__main__":
+    main()
+    import sys
+    import pytest
+
+    sys.exit(pytest.main(["-v", "-s", __file__]))
```
Comment (richardliaw) [inline release/air_tests/air_benchmarks/workloads/gpu_batch_prediction.py:100]: remove this?

File: release/air_tests/air_benchmarks/workloads/gpu_batch_prediction.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+
+@click.command(help="Run Batch prediction on Pytorch ResNet models.")
+@click.option("--data-size-gb", type=int, default=1)
+def main(data_size_gb: int):
+    data_url = f"s3://air-example-data-2/{data_size_gb}G-image-data-synthetic-raw"
+    print(f"Running GPU batch prediction with {data_size_gb}GB data from {data_url}")
+    start = time.time()
+    dataset = ray.data.read_binary_files(paths=data_url)
+    dataset = dataset.map_batches(convert_to_pandas)
```
Comment (ericl) [inline release/air_tests/air_benchmarks/workloads/gpu_batch_prediction.py:60]: Add TODO to use ImageFolder once that merges? (https://github.com/ray-project/ray/pull/24641)

**Review by ericl — APPROVED** [review]

File: release/air_tests/air_benchmarks/workloads/gpu_batch_prediction.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+from torchvision.models import resnet18
+
+import ray
+from ray.air.util.tensor_extensions.pandas import TensorArray
+from ray.train.torch import to_air_checkpoint, TorchPredictor
+from ray.train.batch_predictor import BatchPredictor
+from ray.data.preprocessors import BatchMapper
+
+
+def convert_to_pandas(byte_item_list: List[bytes]) -> pd.DataFrame:
```
Comment (ericl) [inline release/air_tests/air_benchmarks/workloads/gpu_batch_prediction.py:21]: Add TODO to remove once we have ImageFolder.

File: release/air_tests/air_benchmarks/compute_gpu_1.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
+region: us-west-2
+
+max_workers: 0
+
+head_node_type:
+    name: head_node
+    instance_type: g3.8xlarge
+
+worker_node_types:
```
Comment (amogkam) [inline release/air_tests/air_benchmarks/compute_gpu_1.yaml:10]: Remove if this is 0 workers?

**Review by amogkam — APPROVED** [review]

**Comment by amogkam** [comment]
Let’s make sure the release test is passing from the PR before merging this in

**Pull request merged**